### ISSUE TITLE
`Html::tel` removes extension separator

### ISSUE THREAD

**Issue by nilshoerrmann** [getkirby/kirby#5327]
## Description
When formatting a phone number with extension, `Html::tel` removes the extension separator.

**Expected behavior**  
The extension separator should not be removed.

## To reproduce
Take this phone number:  
`+1 416-123-4567 ext. 890`

`Html::tel` produces this `href` value:  
`tel:+14161234567890`

The expected result following RFC3966 would be:
`tel:+1-416-123-4567;ext=890`

This happens because Kirby removes all characters but plus signs and numbers in https://github.com/getkirby/kirby/blob/3.9.5/src/Toolkit/Html.php#L436 and ignores characters allowed by RFC3966. This works well for countries like Germany where extensions are usually not prefixed but it doesn't for countries where this is expected (it also changes phone behaviour when dialing the number – at least I was told, I'm also personally not used to this).

## Your setup
**Kirby Version**  
Kirby 3 and 4.

## Notes

I'm not sure Kirby can solve this without using a huge library like libphonenumber but there should be a note in the docs that both `Html::tel` and `Html::a` as a consequence behave this way. Developers should use `Html::link` instead and pass the correct values by themselves.

**Comment by distantnative** [comment]
Looking at https://datatracker.ietf.org/doc/html/rfc3966 there is indeed a lot else what could be included. I fear you are right that this won't be manageable without a huge library - which we all aren't big fans of. So maybe it really needs to be a note in the docs for these edge cases.

Another option would be to remove the clean-up code line and just use whatever string is passed.

**Comment by nilshoerrmann** [comment]
> Another option would be to remove the clean-up code line and just use whatever string is passed.

This might easily result in invalid href attributes, I think. The current solution is at least valid.

**Comment by distantnative** [comment]
@nilshoerrmann I agree, both aren't ideal. It's deciding between blocking actual valid strings (but which we can't parse) or allowing all strings and putting it in the developers responsibility that they're valid for the tag.

**Comment by nilshoerrmann** [comment]
I've been thinking about this problem again this morning and it is really complex. First off, I think we agree that this should not be a focus of Kirby and thus there should be no library bundled with the core to fix this issue. But: the problem is, without a library this cannot be solved correctly. So in the end, formatting phone numbers according to RFC 3966 is something that should be left to plugins or a custom implementation in a project (most likely relying on libphonenumber as is seems to be the per se standard across many programing languages).

So I think the only purpose of the helper should actually be making sure that the `tel:` prefix is present in the URL. At least if we were starting from scratch.

Thinking of the current helper removing everything but digits: couldn't this just be made optional? That way both options ("digits only" and "RFC-formatted") would be possible. It would be a good compromise between simplicity and validity. Digits only could be the default for compatibility reasons.

**Comment by lukasbestle** [comment]
TBH I think adding another argument makes it quite complex. As an alternative you could prepend `tel:` manually and call `Html::link()` directly.

**Comment by nilshoerrmann** [comment]
@lukasbestle: That's what we do and that's why I suggested to add that to the docs. Still, the method is broken in it's current form. This was just a suggestion to improve the situation without breaking changes.
